.github/workflows/release.yml

```diff
@@ -0,0 +1,60 @@
+name: "Bump version, build, and release"
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  # disable keyring (https://github.com/actions/runner-images/issues/6185):
+  PYTHON_KEYRING_BACKEND: keyring.backends.null.Keyring
+
+jobs:
+  docs:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install poetry
+        run: pipx install poetry
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+          cache: 'pip'
+          cache-dependency-path: 'poetry.lock'
+      - name: Install dependencies
+        run: poetry install
+      - name: Build and push docs
+        run: poetry run mkdocs gh-deploy --force
+  release:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    concurrency: release
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Install poetry
+        run: pipx install poetry
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+          cache: 'pip'
+          cache-dependency-path: 'poetry.lock'
+      - name: Install versioning plugin
+        run: poetry self add "poetry-dynamic-versioning[plugin]"
+      - name: Install dependencies
+        run: poetry install
+      - name: Set Git username
+        run: git config --global user.name "semantic-release (via GitHub Actions)"
+      - name: Set Git email
+        run: git config --global user.email "semantic-release@github-actions"
+      - name: Build and release package
+        env:
+          GH_TOKEN: ${{ secrets.GH_TOKEN }}
+          REPOSITORY_PASSWORD: ${{ secrets.REPOSITORY_PASSWORD }}
+          REPOSITORY_URL: ${{ secrets.REPOSITORY_URL }}
+          REPOSITORY_USERNAME: ${{ secrets.REPOSITORY_USERNAME }}
```

Comment:
These have to be passed accordingly from GitHub settings as secrets / env vars.
Bear in mind that `GH_TOKEN` should have sufficient permissions to push, as per the [docs](https://python-semantic-release.readthedocs.io/en/latest/automatic-releases/github-actions.html).